src/core/qgstemporalnavigationobject.h

```diff
@@ -0,0 +1,157 @@
+/***************************************************************************
+                         qgstemporalnavigationobject.h
+                         ---------------
+    begin                : March 2020
+    copyright            : (C) 2020 by Samweli Mwakisambwe
+    email                : samweli at kartoza dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#ifndef QGSTEMPORALNAVIGATIONOBJECT_H
+#define QGSTEMPORALNAVIGATIONOBJECT_H
+
+#include "qgsrange.h"
+#include <QList>
+#include "qgstemporalcontroller.h"
+
+class QgsMapLayer;
+
+/**
+ * \ingroup core
+ * The QgsTemporalNavigationObject class
+ *
+ * \since QGIS 3.14
+ */
+class CORE_EXPORT QgsTemporalNavigationObject : public QgsTemporalController
+{
+
+  public:
+
+    /**
+      * Constructor for QgsTemporalNavigationObject
+      *
+      */
+    QgsTemporalNavigationObject();
+
+    //! Represents the current VCR navigation status
+    enum NavigationStatus
+    {
+      Forward, //! When the forward button is clicked
+      BackWard, //! When the back button is clicked
+      Next, //! When the next button is clicked
+      Previous, //! When the previous button is clicked
+      Idle //! When no button has been clicked
+    };
+
+    //! Represents the VCR modes in getting data from layers
+    enum Mode
```

Comment:
Mode handling is a per layer setting -- this should be moved to QgsRasterLayerTemporalProperties. You may need different modes for each individual layer in a single project.